### PR TITLE
add data to state

### PR DESCRIPTION
**Pull request by Rahandi**

### CODE REVIEW

File: gocialite.go [hunk 1]
```diff
@@ -136,7 +144,10 @@ func (g *Gocial) Redirect(clientID, clientSecret, redirectURL string) (string, e
 		Endpoint:     endpointMap[g.driver],
 	}
 
-	return g.conf.AuthCodeURL(g.state), nil
+	stateWithData := g.state + "|" + additionaldata
+	stateWithDataEncrypted := base64.StdEncoding.EncodeToString([]byte(stateWithData))
```
Comment (rhzs) [inline gocialite.go]: bukan `encrypted` tapi `encoded`

Comment (rhzs) [reply gocialite.go]: `stateWithEncodedData`

File: gocialite.go [hunk 2]
```diff
@@ -39,18 +39,26 @@ func (d *Dispatcher) New() *Gocial {
...
 	d.mu.RLock()
 	g, ok := d.g[state]
 	d.mu.RUnlock()
 	if !ok {
-		return nil, nil, fmt.Errorf("invalid CSRF token: %s", state)
+		return nil, nil, "", fmt.Errorf("invalid CSRF token: %s", state)
 	}
-	err := g.Handle(state, code)
+
+	stateWithDataDecoded, err := base64.StdEncoding.DecodeString(state)
```
Comment (rhzs) [inline gocialite.go]: `stateWithDecodedData`

File: gocialite.go [hunk 3]
```diff
@@ -39,18 +39,26 @@ func (d *Dispatcher) New() *Gocial {
...
+	if err != nil {
+		return nil, nil, "", err
+	}
+	stateWithDataDecodedSplitted := strings.Split(string(stateWithDataDecoded), "|")
```
Comment (rhzs) [inline gocialite.go]: langsung diassign saja mas `socialState := strings.Split(string(stateWithDataDecoded), "|")[0]`
line bawahnya bisa dihapus

**Review by rhzs — COMMENTED** [review]
minor changes on variable name, shall we add validation when splitting the decoded state?

File: gocialite.go (hunk 3 above)
Comment (Rahandi) [reply gocialite.go]: hasil splitnya yang index 1 di return mas, isinya additional datanya tadi

Comment (rhzs) [reply gocialite.go]: Ok mas

File: gocialite.go [hunk 4]
```diff
@@ -136,7 +144,10 @@ func (g *Gocial) Redirect(clientID, clientSecret, redirectURL string) (string, e
 		Endpoint:     endpointMap[g.driver],
 	}
 
-	return g.conf.AuthCodeURL(g.state), nil
+	stateWithData := g.state + "|" + additionaldata
+	stateWithDataEncoded := base64.StdEncoding.EncodeToString([]byte(stateWithData))
```
Comment (rhzs) [inline gocialite.go]: `stateWithEncodedData`

**Review by rhzs — COMMENTED** [review]
wrong var name

**Review by rhzs — APPROVED** [review]

**Pull request merged**